apps/api/src/recurring-donation/recurring-donation.controller.ts

```diff
@@ -11,19 +11,20 @@ export class RecurringDonationController {
   constructor(private readonly recurringDonationService: RecurringDonationService) {}
 
   @Post()
-  @Public()
   create(@Body() createRecurringDonationDto: CreateRecurringDonationDto) {
     return this.recurringDonationService.create(createRecurringDonationDto)
   }
 
   @Get()
-  @Public()
+  @Roles({
+    roles: [RealmViewSupporters.role, ViewSupporters.role],
+    mode: RoleMatchingMode.ANY,
+  })
   findAll() {
     return this.recurringDonationService.findAll()
   }
 
   @Get(':id')
-  @Public()
```

Comment:
I don't find reasons to have this endpoint available to authenticated users. Maybe we can restrict it to the admins as well